pkg/sql/alter_table.go

```diff
@@ -535,101 +523,52 @@ func (n *alterTableNode) startExec(params runParams) error {
 			}
 
 		case *tree.AlterTableValidateConstraint:
-			info, err := n.tableDesc.GetConstraintInfo()
-			if err != nil {
-				return err
-			}
 			name := string(t.Constraint)
-			constraint, ok := info[name]
-			if !ok {
+			c, _ := n.tableDesc.FindConstraintWithName(name)
+			if c == nil {
 				return pgerror.Newf(pgcode.UndefinedObject,
 					"constraint %q of relation %q does not exist", t.Constraint, n.tableDesc.Name)
 			}
-			if !constraint.Unvalidated {
+			switch c.GetConstraintValidity() {
+			case descpb.ConstraintValidity_Validated:
+				// Nothing to do.
 				continue
+			case descpb.ConstraintValidity_Validating:
+				return pgerror.Newf(pgcode.ObjectNotInPrerequisiteState,
+					"constraint %q in the middle of being added, try again later", t.Constraint)
+			case descpb.ConstraintValidity_Dropping:
+				return pgerror.Newf(pgcode.ObjectNotInPrerequisiteState,
+					"constraint %q in the middle of being dropped", t.Constraint)
```

Comment:
This is one of these instances where the adding and dropping states are handled more uniformly, to no adverse effect from what I can tell.